chips/sam4l/src/gpio.rs

```diff
@@ -79,6 +80,8 @@ pub enum PeripheralFunction {
 const BASE_ADDRESS: usize = 0x400E1000;
 const SIZE: usize = 0x200;
 
+pub static INTERRUPT_COUNT: AtomicUsize = AtomicUsize::new(0);
```

Comment:
My concerns weren't performance, rather correctness. When I see atomic types, I start thinking aggressively about complex reentrancy paths and possible bugs. If this isn't atomic for reentrancy safety, rather language safety, I'd like to see a comment to that effect indicating that you don't really care about this actually being Atomic [though, the notion that it could only be done with an atomic type is in and of itself a bit odd to me]